library/std/src/io/stdio.rs

```diff
@@ -903,6 +905,46 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Prints the given `str` and reads a [`String`] from
+/// [standard input](Stdin). The trailing newline is stripped.
+/// Gives an error on EOF (end of file).
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(io_input)]
+/// use std::io;
+///
+/// fn main() -> io::Result<()> {
+///     let name = io::input("Enter name: ")?;
+///
+///     println!("Your name is {}!", name);
+///
+///     Ok(())
+/// }
+/// ```
+#[unstable(
+    feature = "io_input",
+    reason = "this function may be replaced with a more general mechanism",
+    issue = "none"
+)]
+pub fn input(prompt: &str) -> Result<String> {
```

Comment:
I am still not a fan of `input("")` to read empty string, it prints even though it is empty and the compiler cannot assume printing have no side-effect so it can't be optimized out and will always be called.

Based on @LukasKalbertodt findings, quite a number of others does not even support printing it out.

Reply:
It could be a macro if it was desired to have a variadic arg count. That way printing no text could be just `prompt!()` and then it would expand into a bare call to some `read_line` function.